packages/core/src/components/tree-view/tree-view.stories.ts

```diff
@@ -12,19 +12,39 @@ storiesOfTreeView(
     readme
   },
   {
-    treeViewDemoTemplate: (collection, onOpenItem, onCloseItem, onClickItem) => html`
-      ${treeViewTemplate({
-        collection,
-        onOpenItem: function (e) {
-          onOpenItem(e.detail, collection => this.collection = collection);
-        },
-        onCloseItem: function (e) {
-          onCloseItem(e.detail, collection => this.collection = collection);
-        },
-        onClickItem: function (e) {
-          onClickItem(e.detail);
-        }
-      })}
+    treeViewDemoTemplate: (collection, onOpenItem, onCloseItem, onClickItem, onFilterInput) => html`
+      <div class="container" style="display: flex; width: 100%;">
+        <div style="flex: 1">
+          ${treeViewTemplate({
+            collection,
+            onOpenItem: function (e) {
+              onOpenItem(e.detail, collection => this.collection = collection);
+            },
+            onCloseItem: function (e) {
+              onCloseItem(e.detail, collection => this.collection = collection);
+            },
+            onClickItem: function (e) {
+              onClickItem(e.detail.path, e.detail.originalEvent, collection => this.collection = collection);
+            }
+          })}
+        </div>
+        <div style="flex: 1">
+          <label for="treeViewSearchInput">Zoek</label>
+          <input
+            type="text"
+            id="treeViewSearchInput"
+            @change=${(e: any) => {
```

Comment:
Ik begrijp niet zo goed waarom ik geen live filtering heb. Er wordt pas gefiltered als ik de input blur. Enig idee?

Reply:
Het React component met `onChange` werkt wel zoals verwacht.

Reply:
Nee heel gek, een `@change=${(e: any) => console.log(e.target.value)}` triggert ook niet en een simpel voorbeeld op https://lit.dev/playground/ lijkt ook alleen een update te doen bij blur.

Reply:
Ik heb het gevonden. Het is niet lit maar React die onverwacht gedrag levert. Lit's `@change` doet een binding naar het native `change` event en die wordt pas afgevuurd als het veld wordt geblurred. Voor "live" interactiviteit is het `input` event: Bij **elke** change wordt het even afgevuurd.

React's `onChange` is eigenlijk `onInput`, er zit ook geen verschil tussen deze twee: https://reactjs.org/docs/dom-elements.html#onchange

React heeft een opinionated "change" even listener: https://reactjs.org/docs/dom-elements.html#onchange.

In lit moeten we gebruik maken van `@input`.

Reply:
Aangepast